src/libraries/System.Text.Json/tests/Common/NumberHandlingTests.cs

```diff
@@ -985,6 +1057,27 @@ private async Task AssertUnsupportedFormatThrows<T>()
             }
         }
 
+#if NETCOREAPP
+        [Fact]
+        public async Task InvalidNumberFormatThrows()
+        {
+            await AssertInvalidNumberFormatThrows<Int128>("170141183460469231731687303715884105728"); // MaxValue + 1
+            await AssertInvalidNumberFormatThrows<Int128>("-170141183460469231731687303715884105729"); // MaxValue - 1
+            await AssertInvalidNumberFormatThrows<Int128>("3.14");
+            await AssertInvalidNumberFormatThrows<UInt128>("340282366920938463463374607431768211456"); // MaxValue + 1
+            await AssertInvalidNumberFormatThrows<Int128>("3.14");
+            await AssertInvalidNumberFormatThrows<UInt128>("-1");
+            await AssertInvalidNumberFormatThrows<Half>("65520");
+            await AssertInvalidNumberFormatThrows<Half>("-65520");
```

Comment:
Per @GrabYourPitchforks feedback could you add a test case that uses leading zeros?

Reply:
This might be a better place to add the test case:
https://github.com/dotnet/runtime/blob/43fe8a8fc7642a1268e80b0433aebb92b049ee09/src/libraries/System.Text.Json/tests/Common/NumberHandlingTests.cs#L1044C28-L1052